encoder/src/main/java/openlr/encoder/data/LocRefPoint.java

```diff
@@ -323,7 +323,8 @@ private double calculateBearing(final OpenLREncoderProperties properties)
         } else {
             dir = GeometryUtils.BearingDirection.AGAINST_DIRECTION;
         }
-        int projection = 0;
+        int projection = (dir == GeometryUtils.BearingDirection.IN_DIRECTION) ? 0 : line.getLineLength();
```

Comment:
To make the decisions more clear I suggest a refactoring like this:
```
    private double calculateBearing(final OpenLREncoderProperties properties) {
        GeometryUtils.BearingDirection dir;
        if (isLast) {
            dir = GeometryUtils.BearingDirection.AGAINST_DIRECTION;
        } else {
            dir = GeometryUtils.BearingDirection.IN_DIRECTION;
        }
        int projection;
        if (isPointOnLine) {
            projection = line.measureAlongLine(longitude, latitude);
        } else if (isLast) {
            projection = line.getLineLength();
        } else {
            projection = 0;
        }
        return GeometryUtils.calculateLineBearing(line, dir, properties.getBearingDistance(), projection);
    }
```

Reply:
0975cd3318040870779bc3de4b706cd2ffb8b2ef